app/Services/Three/PostService.php

```diff
@@ -87,6 +87,31 @@ public function update($post, $data)
         return $post;
     }
 
+    /**
+     * Handles all business logic around reviewing posts.
+     *
+     * @param \Rogue\Models\Signup $signup
+     * @param array $data
+     * @return \Rogue\Models\Post|\Rogue\Models\Signup
+     */
+    public function review($post, $data)
```

Comment:
what is the difference between `$post` and `$data` here? in the other `PostService` we only pass `$data`

Reply:
@sbsmith86 hmm this is a good question. This is what the PostRepository asks for (it actually needs `$comment` too which I've added) so I went with that

Reply:
When possible, I think it's a lot clearer to pass real models and named parameters around, rather than big arrays (where the dependency is not as explicit).

A side benefit is that we can check whether the post exists & use that same instance elsewhere in the request, whereas the `post_id` parameter in the old repository always needed to make a fresh query to load it anew from the DB.